pages/en/lb4/Routes.md

```diff
@@ -48,17 +52,23 @@ const spec = {
     }
   }
 };
-const route = new Route('get', '/', spec, greet);
-app.route(route);
 
-app.start();
+(async function start() {
+  const server = await app.getServer(RestServer);
+  const route = new Route('get', '/', spec, greet);
+  server.route(route);
+  await app.start();
+})();
```

Comment:
This will swallow any errors thrown from `start`, causing "unhandled promise rejection" warnings. You should add `.catch()` handler at minimum.

It makes me wonder, can we move methods like `route`, `handler` and `sequence` to a different class, one that does not have any dependencies so that it will always resolve synchronously, and then inject this class as a RestServer dependency?

An mock-up:

// inside RestComponent (or a mixin?)
app.rest = new RestServerConfigurator();
app.bind(RestBindings.CONFIGURATOR).toDynamicValue(() => app.rest);

// in RestServer constructor
class RestServer {
  constructor(
    @inject(RestBindings.CONFIGURATOR) configurator: RestServerConfigurator,
    /*...*/) {
      // define routes, handler, sequence etc. from the configurator
    }
 }

// usage in applications
app.rest.route('get', '/', spec, greet);
```

This can be extended to support multiple rest servers by replacing `app.rest` with `app.${serverName}` , where the server name is the key used in `servers.${serverName}` binding and in `appConfig.servers` object.

Thoughts?

Reply:
Let's leave this out of scope of this documentation update.